src/app/drive/components/ItemDetailsDialog/ItemDetailsDialog.tsx

```diff
@@ -0,0 +1,198 @@
+import ItemDetailsSkeleton from './components/ItemDetailsSkeleton';
+
+const Header = ({ title, onClose }: { title: string; onClose: () => void }) => {
+  return (
+    <>
+      <span
+        className="max-w-full overflow-hidden overflow-ellipsis whitespace-nowrap text-xl font-medium"
+        title={title}
+      >
+        {title}
+      </span>
+      <div className="flex h-9 w-9 cursor-pointer items-center justify-center rounded-md bg-black bg-opacity-0 transition-all duration-200 ease-in-out hover:bg-opacity-4 active:bg-opacity-8">
+        <X onClick={onClose} size={22} />
+      </div>
+    </>
+  );
+};
+
+const ItemsDetails = ({ item, translate }: { item: ItemDetailsProps; translate: (key: string) => string }) => {
+  return (
+    <>
+      {Object.entries(item).map(([key, value]) => {
+        if (!value) return;
+        return (
+          <div key={key} className="flex w-full max-w-xxxs flex-col items-start justify-center space-y-0.5">
+            <p className="text-sm font-medium text-gray-50">
+              {translate(`modals.itemDetailsModal.itemDetails.${key}`)}
+            </p>
+            <p title={value} className="w-full text-base font-medium text-gray-100 line-clamp-2">
+              {value}
+            </p>
+          </div>
+        );
+      })}
+    </>
+  );
+};
+
+/**
+ * Return all the details of the item selected
+ * The data is:
+ * - Name
+ * - Shared
+ * - Size (only for files)
+ * - Type (only for files)
+ * - Uploaded
+ * - Modified
+ * - Uploaded by
+ * - Location
+ *  */
+
+const ItemDetailsDialog = ({ onSharedFolderClicked }: { onSharedFolderClicked?: (item: any) => void }) => {
+  const dispatch = useAppDispatch();
+  const isOpen = useAppSelector((state: RootState) => state.ui.isItemDetailsDialogOpen);
+  const item = useAppSelector((state: RootState) => state.ui.itemDetails);
+  const { translate } = useTranslationContext();
+  const [itemProps, setItemProps] = useState<ItemDetailsProps>();
+  const [isLoading, setIsLoading] = useState(false);
+  const IconComponent = iconService.getItemIcon(item?.type === 'folder', item?.type);
+  const itemName = `${item?.plainName ?? item?.name}` + `${item?.type && !item.isFolder ? '.' + item?.type : ''}`;
+  const user = localStorageService.getUser();
+  const { onNameClicked } = useDriveItemActions(item as DriveItemData);
+  const isFolder = item?.isFolder;
+
+  useEffect(() => {
+    if (isOpen && item && user) {
+      setIsLoading(true);
+      const isShared = item.isShared ? translate('actions.yes') : translate('actions.no');
+      const uploaded = formateDate(item.createdAt);
+      const modified = formateDate(item.updatedAt);
+      getDetailsData(item, isShared, uploaded, modified, user.email)
+        .then((details) => {
+          setItemProps(details);
+          setIsLoading(false);
+        })
+        .catch((err) => {
+          const error = errorService.castError(err);
+          errorService.reportError(error);
+        });
```

Comment:
If something during `getDetailsData` execution fails, this catch never be executed because already exists a try catch block inside the function. 
Just to keep in mind, the behaviour if something goes wrong will be that the detail block is not displayed, is that behaviour correct?